### PR TITLE
Adding the data provider annotation to Unit Tests

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) adds a new page with additional information related to **dataProvider** annotation that is widely used on writing Unit Tests.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/test/unit/annotations/data-provider.html
- https://devdocs.magento.com/guides/v2.2/test/unit/annotations/data-provider.html
- https://devdocs.magento.com/guides/v2.3/test/unit/annotations/data-provider.html

whatsnew
Added the [Data Provider Annotation](https://devdocs.magento.com/guides/v2.3/test/unit/annotations/data-provider.html) topic to the PHP Unit Testing guide.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/test/unit/annotations/data-provider.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+group: testing
+title: Data Provider Annotation
+---
+
+A Data Provider allows you to define the test once and run it multiple times with different inputs, by checking multiple use cases.
```
Comment (rogyar) [inline guides/v2.1/test/unit/annotations/data-provider.md]: Please, adjust
```suggestion
A Data Provider allows you to define the test once and run it multiple times with different inputs, for checking multiple use cases.
```

Comment (eduard13) [reply guides/v2.1/test/unit/annotations/data-provider.md]: @rogyar, adjusted.

**Comment by rogyar** [comment]
Examples from the codebase:
https://github.com/magento/magento2/blob/2.3/app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php#L86
https://github.com/magento/magento2/blob/2.2/app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php#L186
https://github.com/magento/magento2/blob/2.1/app/code/Magento/Rule/Test/Unit/Model/Condition/CombineTest.php#L67

**Pull request merged**